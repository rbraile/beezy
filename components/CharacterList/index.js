import React from 'react'
import orderBy from 'lodash/orderBy'
import Card from '../../components/Card'
import Filters from '../../components/Filters'
import Order from '../../components/Order'

import {
  CharacterListContainer,
  CharacterSection,
  Aside,
  Container,
} from './styled'
import Breadcrumb from '../Breadcrumb'

const orden = (charactersList, order) => {
  if (order !== {}) {
    return orderBy(charactersList, [`${order.column}`], [`${order.type}`])
  }
}

const CharacterList = ({
  charactersList,
  handlerSetOrder,
  order,
  handlerSearch,
  search,
  clearList,
}) => {
  return (
    <CharacterSection>
      <Breadcrumb />
      <h1>List of Star Wars characters</h1>

      <Container>
        <Aside>
          {search !== '' && (
            <p>
              Search: <strong>{search}</strong>
            </p>
          )}
          <Filters clearList={clearList} handlerSearch={handlerSearch} />
          <Order handlerSetOrder={handlerSetOrder} order={order} />
        </Aside>
        <CharacterListContainer>
          {!charactersList.length ? (
            <p>
              There is no results: <strong>{search}</strong>
            </p>
          ) : (
            orden(charactersList, order).map(character => (
              <Card key={character.name} {...character} />
            ))
          )}
        </CharacterListContainer>
      </Container>
    </CharacterSection>
  )
}

export default CharacterList
