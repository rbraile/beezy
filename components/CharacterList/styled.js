import styled from 'styled-components'
import device from '../../utils/device'

export const CharacterListContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 74%;

  @media ${device.tablet} {
    width: 100%;
  }
`

export const CharacterSection = styled.section``

export const Container = styled.section`
  display: flex;
  justify-content: space-around;

  @media ${device.tablet} {
    flex-direction: column;
  }
`

export const Aside = styled.div`
  padding: 10px;
  background: #eee;
  width: 23%;

  @media ${device.tablet} {
    width: 100%;
    margin-bottom: 20px;
  }
`

export const TitlePage = styled.h1`
  padding: 20px;
  text-align: center;
  font-size: 20px;
`
