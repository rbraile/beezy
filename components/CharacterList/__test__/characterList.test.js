import React from 'react'
import { shallow, mount } from 'enzyme'

import CharacterList from '../index'

describe('TEST = CHARACTERLIST', () => {
  const handlerSetOrder = jest.fn()
  const handlerSearch = jest.fn()
  const clearList = jest.fn()

  const charactersList = {
    results: [],
  }
  const order = { column: 'name', type: 'asc' }
  test('TEST = CHARACTERLIST = SNAPSHOT', () => {
    const wrapper = shallow(
      <CharacterList
        handlerSetOrder={handlerSetOrder}
        charactersList={charactersList.results}
        order={order}
        handlerSearch={handlerSearch}
        search=''
        clearList={clearList}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('TEST = CHARACTERLIST = Check search param', () => {
    const charactersList = {
      results: [
        {
          name: 'Luke Skywalker',
          height: '172',
          mass: '77',
          hair_color: 'blond',
          skin_color: 'fair',
          eye_color: 'blue',
          birth_year: '19BBY',
          gender: 'male',
        },
      ],
    }

    const wrapper = shallow(
      <CharacterList
        handlerSetOrder={handlerSetOrder}
        charactersList={charactersList.results}
        order={order}
        handlerSearch={handlerSearch}
        clearList={clearList}
      />
    )

    expect(wrapper.find('Card')).toHaveLength(1)
  })

  test('TEST = CHARACTERLIST = Check search param', () => {
    const searchWord = 'yoda'
    const wrapper = shallow(
      <CharacterList
        handlerSetOrder={handlerSetOrder}
        charactersList={charactersList.results}
        order={order}
        handlerSearch={handlerSearch}
        search={searchWord}
        clearList={clearList}
      />
    )

    const search = wrapper
      .find('styled__Aside')
      .find('strong')
      .text()

    expect(search).toBe('yoda')
  })
})
