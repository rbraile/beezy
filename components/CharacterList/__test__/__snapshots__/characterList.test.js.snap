// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST = CHARACTERLIST TEST = CHARACTERLIST = SNAPSHOT 1`] = `
<styled__CharacterSection>
  <Breadcrumb />
  <h1>
    List of Star Wars characters
  </h1>
  <styled__Container>
    <styled__Aside>
      <Filters
        clearList={[MockFunction]}
        handlerSearch={[MockFunction]}
      />
      <Order
        handlerSetOrder={[MockFunction]}
        order={
          Object {
            "column": "name",
            "type": "asc",
          }
        }
      />
    </styled__Aside>
    <styled__CharacterListContainer>
      <p>
        There is no results: 
        <strong />
      </p>
    </styled__CharacterListContainer>
  </styled__Container>
</styled__CharacterSection>
`;
